src/main/java/com/ericsson/gerrit/plugins/eiffel/state/State.java

```diff
@@ -0,0 +1,131 @@
+    protected static final Logger LOGGER = LoggerFactory.getLogger(State.class);
+    protected static final String FILE_ENDING = "db";
+
+    protected File pluginDir;
+
+    public State(File pluginDir) {
+        this.pluginDir = pluginDir;
+    }
+
+    public abstract String getEventId(String project, String tableColumnName) throws NoSuchElementException, ConnectException, FileNotFoundException;
+
+    public abstract void setState(String eiffelEventId, EiffelEvent eiffelEvent)
+            throws NoSuchElementException, ConnectException, SQLException;
+
+    protected String getLastSubmittedEiffelEvent(String project, String tableColumnName,
+            Table tableName)
+            throws NoSuchElementException, ConnectException, FileNotFoundException {
+        try {
+            File parentDir = new File(buildParentFilePath(pluginDir, project));
+            if (!(parentDir.exists())) {
+                throw new FileNotFoundException(parentDir.getName() + " could not be created.");
+            }
+
+            return getEventId(project, tableColumnName, tableName);
+        } catch (NoSuchElementException e) {
+            throw new NoSuchElementException(
+                    "Database did not return any value for this query\n" + "Exception Message:" + e.getMessage());
+        }
+
+    }
+
+    protected void setLastSubmittedEiffelEvent(String project, String tableColumnName,
+            String eiffelEvent,
+            Table tableName) throws NoSuchElementException, ConnectException, SQLException {
+        String fileName = String.format("%s.%s", project, FILE_ENDING);
+        DataBaseHandler dBHandler = new DataBaseHandler(pluginDir, fileName);
+        String parentPath = buildParentFilePath(pluginDir, project);
+        createParentDirsIfNecessary(parentPath);
+
+        saveEventToDatabase(project, tableColumnName, eiffelEvent, tableName, dBHandler);
+    }
+
+    private void saveEventToDatabase(String project, String tableColumnName, String eiffelEvent,
+            Table tableName, DataBaseHandler dBHandler) throws ConnectException, SQLException {
+        String oldEvent = getOldEventId(dBHandler, tableName, tableColumnName);
+
+        if (!oldEvent.isEmpty()) {
+            dBHandler.updateInto(tableName, tableColumnName, eiffelEvent);
+            LOGGER.info("Replaced old event id '{}' with new event if '{}', for project '{}', and branch '{}'.",
+                    oldEvent, eiffelEvent, project, tableColumnName);
+        } else {
+            dBHandler.insertInto(tableName, tableColumnName, eiffelEvent);
+            LOGGER.info("Saved eiffel event with id '{}', for project '{}', and branch '{}'.", eiffelEvent, project,
+                    tableColumnName);
+        }
+    }
+
+    /**
+     * Creates parent directories of a project if they don't exist and is included
+     * in the project name.
+     *
+     * @param path
+     */
+    protected void createParentDirsIfNecessary(String path) {
```

Comment:
State should maybe not handle creation of path, this should be closer to the DB handler